src/Microsoft.AspNet.Mvc.Core/OutputFormatterOptions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    public class OutputFormatterOptions
+    {
+        private Dictionary<string, MediaTypeHeaderValue> map = new Dictionary<string, MediaTypeHeaderValue>();
+
+        public void AddFormatMapping(string format, MediaTypeHeaderValue contentType)
+        {
+            if (!string.IsNullOrEmpty(format) && contentType != null)
+            {
+                if(format.StartsWith("."))
+                {
+                    format = format.TrimStart('.');
```

Comment:
If just returns the same string if it doesn't start with period, for second case it will remove all the dots.